### PR TITLE
API-13714 remove observation dstu2 query hack

### PR DESCRIPTION
**Pull request by jblefkove-va**

### CODE REVIEW

File: data-query/src/test/java/gov/va/api/health/dataquery/service/controller/observation/Dstu2ObservationControllerTest.java [hunk 1]
```diff
@@ -259,86 +234,64 @@ void searchByPatient() {
   }
 
   @Test
-  void searchByPatientAndCategoryAndOneDateHack() {
-    /*
```
Comment (taylorrassmannlibertyits) [inline data-query/src/test/java/gov/va/api/health/dataquery/service/controller/observation/Dstu2ObservationControllerTest.java:263]: Why did you remove this comment block?

Comment (jblefkove-va) [reply data-query/src/test/java/gov/va/api/health/dataquery/service/controller/observation/Dstu2ObservationControllerTest.java:263]: The query-hack-specific tests were removed

Comment (taylorrassmannlibertyits) [reply data-query/src/test/java/gov/va/api/health/dataquery/service/controller/observation/Dstu2ObservationControllerTest.java:263]: Ok, but the other non hacks had a comment block too. I'm ok with the removal, just wanted to confirm.

Comment (jblefkove-va) [reply data-query/src/test/java/gov/va/api/health/dataquery/service/controller/observation/Dstu2ObservationControllerTest.java:263]: It is not very clear in the diff, but `searchByPatientAndCategory()` (originally `searchByPatientAndCategoryNoHack()`) has no comment block, while `searchByPatientAndCategoryAndOneDate()` (originally `searchByPatientAndCategoryAndOneDateNoHack()`) does

**Review by taylorrassmannlibertyits — APPROVED** [review]
